src/PostgREST/ApiRequest.hs

```diff
@@ -123,6 +124,8 @@ userApiRequest schema req reqBody =
               Nothing -> PayloadParseError "All lines must have same number of fields"
               Just json -> PayloadJSON json)
             (CSV.decodeByName reqBody)
+        Right OpenAPI ->
+          PayloadParseError "Content-type not acceptable: application/openapi+json; charset=utf-8"
```

Comment:
To avoid repeating the mime type you could do

```haskell
Right oa@OpenAPI ->
  PayloadParseError $ "Content-type not acceptable: " <> cs (show oa)
```